### ISSUE TITLE
metadata grpc deployment pod is in CrashLoopBackOff state

### ISSUE THREAD

**Issue by satish-dops94** [awslabs/kubeflow-manifests#149]
Hi All, I am actually new to kubeflow, I followed the ReadMe section and deployed the kubeflow with RDS and S# connectivity.

**Description**
After the deployment, everything was working fine apart from the RDS connectivity with the metadata grpc deployment is not working properly, due to which ml-pipeline is not up and running. 

Below are the logs from the pod:
**metadata-grpc-deployment:**
```
2022-03-19 19:34:59.167563: F ml_metadata/metadata_store/metadata_store_server_main.cc:220] Non-OK-status: status status: Internal: mysql_real_connect failed: errno: 1045, error: Access denied for user 'YOUR_RDS_USERNAME'@'<Ip-address>' (using password: YES)MetadataStore cannot be created with the given connection config.
```

**ml-pipeline:**
```
I0319 19:35:20.040246       8 config.go:57] Config DBConfig.ExtraParams not specified, skipping
```

I tried connecting to the RDS instance from the kubeflow namespace, but I am getting access denied. However when I executed the same command from the default or some other kube-system namespace I was able to enter it into MySQL.
```
kubectl run -it --rm --image=mysql:5.7 --restart=Never mysql-client -- mysql -h <YOUR RDS ENDPOINT> -u <YOUR LOGIN> -p<YOUR PASSWORD>
```

**Additional Info:**

- I have cross-checked the AWS secrets-manager too, and they are being populated correctly. 

**Version:**
kubeflow version: 1.3
Kubernetes version: 1.20
kubectl version: 1.23

**Comment by satish-dops94** [comment]
I was able to resolve the above issue. I believe the culprit was the go package version which installed the kustomize.
But I am facing another use while bringing up the ml-pipeline pod:

```
I0321 18:11:16.495688       8 client_manager.go:155] Initializing client manager
I0321 18:11:16.495775       8 config.go:57] Config DBConfig.ExtraParams not specified, skipping
F0321 18:11:16.774731       8 client_manager.go:400] Failed to check if Minio bucket exists. Error: Access Denied
```

Any suggestions, as if this is the known issue while trying to connect to the S3 bucket.

**Comment by rrrkharse** [comment]
Hi @satish-dops94, I'll need to look deeper into the issue but looks like some configurations were not applied appropriately. This can happen when a `kustomize` version > 3.20 is used. Are you using `kustomize` to apply the manifests and are you using version 3.20?
```
$ kustomize version

Version: {KustomizeVersion:3.2.0 GitCommit:a3103f1e62ddb5b696daa3fd359bb6f2e8333b49 BuildDate:2019-09-18T16:26:36Z GoOs:darwin GoArch:amd64}
```

**Comment by satish-dops94** [comment]
Hi @rrrkharse, Few points I would like to mention, I was also thinking this might have caused some issue but the below points drove me into confusion. 

- I have installed the Kustomize from the source code 3.2.0 only, but the go package installed in my system was using GoArch:amd64, and my OS architecture is GoArch:arm64.
- I installed the whole kubeflow setup using the amd64 version of the go package, due to which pipelines patch wasn't applied properly. So I had to uninstall my amd64 go package and install the arm64 go package and then re-applied the kubeflow pipelines with Kustomize 3.2.0 which resolve my very initial query of this issue.
- While I was checking the version of Kustomize, I get the below output:
```
~/go/bin/kustomize version                                      
Version: {KustomizeVersion:unknown GitCommit:$Format:%H$ BuildDate:1970-01-01T00:00:00Z GoOs:darwin GoArch:arm64}
```
- Also I was wondering, if should I delete the setup and re-install the whole cluster with GoArch:arm64 Kustomize. Would this resolve the error as it did for the kubeflow pipelines?

**Comment by goswamig** [comment]
@satish-dops94  couple of questions

what platform u r running the kubeflow on EKS or something else?? Is it EKS or something else? 
Whats your gateway from which you're deploying kubeflow (your Mac or window or something else?)

**Comment by satish-dops94** [comment]
Hi @rrrkharse, @goswamig, Thanks for your responses and support..!!
@goswamig, to answer your question, I am running kubeflow on EKS and I am deploying kubeflow from MAC onto the Linux versions only.

I was able to fix the issue, by re-installing the secrets-manager and kubeflow pipelines deployments, but before installing I had to delete the existing deployment completely and again had done a fresh installation of the above deployments. Though I wasn't able to conclude what has caused the Access denied error. I need to re-visit the whole installation setup on a fresh cluster to reproduce the error or confirm the proper fix. 

In short, further investigation is still going on to find the root cause of this behavior.

**Comment by ryansteakley** [comment]
We have this issue in our backlog to investigate any root causes, currently we are not able to reproduce this issue and the user has been to able to successfully setup.
